### PR TITLE
akka-persistence compiling on Scala3

### PR DESCRIPTION
**Pull request by ignasi35**
References #30243

### CODE REVIEW

**Comment by ignasi35** [comment]
A couple of open front in main compile (haven't dug into `test:compile` yet). Main issue is on the java code  generated by `protoc` for `MessageFormats`:

```
[error] -- [E049] Reference Error: /Users/ignasi/git/github/akka/akka/akka-persistence-scala3/akka-persistence/src/main/java/akka/persistence/serialization/MessageFormats.java:4099:13 
[error] 4099 |      public Builder newBuilderForType() { return newBuilder(); }
[error]      |             ^^^^^^^
[error]      |Reference to Builder is ambiguous,
[error]      |it is both defined in object AtLeastOnceDeliverySnapshot
[error]      |and imported subsequently by import akka.persistence.serialization.MessageFormats.AtLeastOnceDeliverySnapshot.
[error]      |  UnconfirmedDelivery
[error]      |._
```

**Comment by ignasi35** [comment]
>  Main issue is on the java code generated by `protoc` for `MessageFormats`: ...

I've confirmed offline with the scala team this is a scala 3 bug. Will need to do a few more tests and check a few suggestions for workarounds.

File: akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbIdMapping.scala [hunk 1]
```diff
@@ -66,7 +66,7 @@ private[persistence] trait LeveldbIdMapping extends Actor { this: LeveldbStore =
   private def writeIdMapping(id: String, numericId: Int): Int = {
     idMap = idMap + (id -> numericId)
     leveldb.put(keyToBytes(mappingKey(numericId)), id.getBytes(UTF_8))
-    newPersistenceIdAdded(id)
+    this.newPersistenceIdAdded(id)
```
Comment (ignasi35) [inline akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbIdMapping.scala:69]: This change I'm not too sure either.

The compiler error is:

```

[error] -- [E049] Reference Error: /Users/ignasi/git/github/akka/akka/akka-persistence-scala3/akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbIdMapping.scala:69:4 
[error] 69 |    newPersistenceIdAdded(id)
[error]    |    ^^^^^^^^^^^^^^^^^^^^^
[error]    |    Reference to newPersistenceIdAdded is ambiguous,
[error]    |    it is both defined in trait LeveldbIdMapping
[error]    |    and inherited subsequently in trait LeveldbIdMapping

```

And the cause is:

```
private[persistence] trait LeveldbIdMapping extends Actor { this: LeveldbStore =>
```

where `LeveldbStore` extends `LeveldbIdMapping` overriding `newPersistenceIdAdded`.

All this code is on the LevelDb journal which is deprecated (https://github.com/akka/akka/pull/30213) and it's `private[akka]`. I'm tempted to merge the two traits `LeveldbIdMapping` and `LeveldbStore` into one.

**Comment by ignasi35** [comment]
Rebased on `master`

**Review by raboof — APPROVED** [review]

**Comment by andreaTP** [comment]
LGTM

**Review by andreaTP — APPROVED** [review]

**Pull request merged**